cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1225,6 +1227,99 @@ public void getInstances(FullHttpRequest request, HttpResponder responder,
+   * possible data:
+   * </p>
+   * <pre><code>
+   * [{"appId": "App1", "programType": "Service", "programId": "Service1",
+   * "statusCode": 200, "runCount": 20},
+   * {"appId": "App1", "programType": "Workflow", "programId": "testWorkflow", "statusCode": 404,
+   * "error": "Program 'testWorkflow' is not found"},
+   *  {"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow", "runnableId": "Flowlet1",
+   *  "statusCode": 200, "runCount": 300}]
+   * </code></pre>
+   */
+  @POST
+  @Path("/runcount")
+  public void getRunCounts(FullHttpRequest request, HttpResponder responder,
+                           @PathParam("namespace-id") String namespaceId) throws Exception {
+    List<BatchProgram> programs = validateAndGetBatchInput(request, BATCH_PROGRAMS_TYPE);
+    if (programs.size() > 100) {
+      throw new BadRequestException(String.format("%s programs found in the request, the maximum number " +
+                                                    "supported is 100", programs.size()));
+    }
+
+    List<ProgramId> programIds =
+      programs.stream().map(batchProgram -> new ProgramId(namespaceId, batchProgram.getAppId(),
+                                                          batchProgram.getProgramType(),
+                                                          batchProgram.getProgramId())).collect(Collectors.toList());
+    List<BatchProgramCount> counts = new ArrayList<>(programs.size());
+    for (RunCountResult runCountResult : lifecycleService.getProgramRunCounts(programIds)) {
+      ProgramId programId = runCountResult.getProgramId();
+      Exception exception = runCountResult.getException();
+      if (exception == null) {
+        counts.add(new BatchProgramCount(programId, 200, null, runCountResult.getCount()));
+      } else if (exception instanceof NotFoundException) {
+        counts.add(new BatchProgramCount(programId, 404, exception.getMessage(), null));
+      } else if (exception instanceof UnauthorizedException) {
+        counts.add(new BatchProgramCount(programId, 403, exception.getMessage(), null));
+      }
+    }
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(counts));
+  }
+
+  /**
+   * Returns the count of the given program runs
+   */
+  @GET
+  @Path("/apps/{app-name}/{program-type}/{program-name}/runcount")
+  public void getProgramRunCount(HttpRequest request, HttpResponder responder,
+                                 @PathParam("namespace-id") String namespaceId,
+                                 @PathParam("app-name") String appName,
+                                 @PathParam("program-type") String type,
+                                 @PathParam("program-name") String programName) throws Exception {
+    ProgramType programType = getProgramType(type);
+    ProgramId programId = new ProgramId(namespaceId, appName, programType, programName);
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(lifecycleService.getProgramRunCount(programId)));
```

Comment:
Call the other `getProgramRunCount` method by providing `ApplicationId.DEFAULT_VERSION` as the app version